drivers/flash/spi_nor.c

```diff
@@ -576,55 +610,64 @@ static int mxicy_rdcr(const struct device *dev)
  *
  * @return 0 on success or a negative error code.
  */
-static int mxicy_wrcr(const struct device *dev,
-			uint16_t cr)
+static int mxicy_wrcr(const struct device *dev, uint16_t cr)
 {
+	const struct spi_nor_config *cfg = dev->config;
+	int ret = -ENOSYS;
 	/* The configuration register bytes on the Macronix MX25R devices are
 	 * written using the Write Status Register command where the configuration
 	 * register bytes are written as two extra bytes after the status register.
 	 * First read out the current status register to preserve the value.
 	 */
-	int sr = spi_nor_rdsr(dev);
 
-	if (sr < 0) {
-		LOG_ERR("Read status register failed: %d", sr);
-		return sr;
-	}
+	if (cfg->mxicy_mx25r_power_mode_exist) {
+		int sr = spi_nor_rdsr(dev);
 
-	int ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_WREN);
+		if (sr < 0) {
+			LOG_ERR("Read status register failed: %d", sr);
+			return sr;
+		}
 
-	if (ret == 0) {
-		uint8_t data[] = {
-			sr,
-			cr & 0xFF,	/* Configuration register 1 */
-			cr >> 8		/* Configuration register 2 */
-		};
+		ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_WREN);
+
+		if (ret == 0) {
+			uint8_t data[] = {
+				sr, cr & 0xFF, /* Configuration register 1 */
+				cr >> 8		/* Configuration register 2 */
+			};
 
 		ret = spi_nor_access(dev, SPI_NOR_CMD_WRSR, NOR_ACCESS_WRITE, 0, data,
 			sizeof(data));
 		spi_nor_wait_until_ready(dev);
+		}
 	}
 
 	return ret;
 }
 
 static int mxicy_configure(const struct device *dev, const uint8_t *jedec_id)
 {
+	const struct spi_nor_config *cfg = dev->config;
+	int ret = -ENOSYS;
+
+	if (cfg->mxicy_mx25r_power_mode_exist) {
 	/* Low-power/high perf mode is second bit in configuration register 2 */
-	enum { LH_SWITCH_BIT = 9 };
+		enum {
+			LH_SWITCH_BIT = 9
+		};
 	const uint8_t JEDEC_MACRONIX_ID = 0xc2;
 	const uint8_t JEDEC_MX25R_TYPE_ID = 0x28;
-	int current_cr, new_cr, ret;
+		int current_cr, new_cr;
 	/* lh_switch enum index:
 	 * 0: Ultra low power
 	 * 1: High performance mode
 	 */
-	const bool use_high_perf = DT_INST_ENUM_IDX(0, mxicy_mx25r_power_mode);
+		const bool use_high_perf = cfg->mxicy_mx25r_power_mode;
```

Comment:
Marker when review stopped - can you review the comments I've left so far and apply them to the whole file? I am finding reviewing this a bit tedious because lots of the formatting changes were not needed and are not good, it makes reading the file much harder and seeing what your changes are vs what is a formatting change isn't straight forward.